src/modules/dashboard/sagas/utils/safeHelpers.ts

```diff
@@ -280,6 +281,92 @@ export const getTransferFundsData = async (
   ]);
 };
 
+const getParsedJSONOrArrayFromString = (
+  parameter: string,
+): (string | number)[] | null => {
+  try {
+    const arrayResult = JSON.parse(parameter);
+    return arrayResult.map((value) => {
+      if (Number.isInteger(value)) {
+        return new BigNumber(value).toString();
+      }
+      return value;
+    });
+  } catch (err) {
+    return null;
+  }
+};
+
+const isArrayParameter = (parameter: string): boolean =>
+  /(\[\d*])+$/.test(parameter);
```

Comment:
How come you're only checking for arrays containing numeric digits? If I understand correctly, this is intended to test whether the start of the string is '[' and the end ']' (i.e. an array)? In which case, the following might be simpler: 
`parameter[0] + parameter[parameter.length - 1] === '[]'`

Reply:
You are right, I don't remember why did I include the `\d*`